include/mbedtls/mbedtls_config.h

```diff
@@ -3206,6 +3206,24 @@
  */
 #define MBEDTLS_PSA_CRYPTO_C
 
+/**
+ * \def MBEDTLS_PSA_STATIC_KEY_SLOTS
+ *
+ * Statically preallocate all key slot buffers to store volatile keys in PSA
+ * instead of allocating them dynamically when required. This helps reducing
+ * heap memory usage as well as heap management code's footprint in embedded
+ * devices.
+ *
+ * \note This feature comes with a (potentially) higher RAM usage since:
+ *       - All the key slots are allocated no matter if they are used of not.
+ *       - Each key slot's length is as large as the largest key type supported
+ *         in the build.
+ *
+ * Requires: MBEDTLS_PSA_CRYPTO_C
+ *
+ */
+//#define MBEDTLS_PSA_STATIC_KEY_SLOTS
```

Comment:
I'm not sure if it's technically possible to have both this and [dynamic key slots](https://github.com/Mbed-TLS/mbedtls/pull/9240), but we should exclude it because it would be additional testing burden and it would make no sense.

We also need to make the naming non-confusing, since there will be three states: dynamically allocated key slices (for the A-class world), statically allocated key slices but dynamically allocated key material (for when RAM and code size are limited but there's still a heap), and fully static (for going without a heap).

Reply:
> I'm not sure if it's technically possible to have both this and https://github.com/Mbed-TLS/mbedtls/pull/9240, but we should exclude it because it would be additional testing burden and it would make no sense.

Oh, I wasn't aware of https://github.com/Mbed-TLS/mbedtls/pull/9240, but I agree that the build option/feature added there is in contrast to the one added here, so it makes no sense to enabled them both at the same time. Since this PR need a design approval while yours is on review, I guess that 9240 will likely be merged before this one. Once that happens I will rebase this PR on top of `development` and do the proper changes.

> We also need to make the naming non-confusing, since there will be three states

No problem at all on this, I just pick something that made sense to me, but I'm completely open to new/better ideas.